woocommerce/woocommerce-gateway-stripe
# [ECE] Hide express checkout if cards are disabled

Fixes #3546 

## Changes proposed in this Pull Request:
When credit cards are not enabled, using Apple Pay/Google Pay will result in an `Invalid payment method` error. This is because [checkout fails validation due to the `stripe` payment gateway not being available](https://github.com/woocommerce/woocommerce/blob/58a59a04b9e60de2ad1f4bfda57b466509a317e7/plugins/woocommerce/includes/class-wc-checkout.php#L973).

In this PR, when credit cards are not enabled, we will not display Apple Pay/Google Pay for block cart and block checkout, as well as the wp-admin Payment Methods page. This is already the behavior for shortcode checkout, shortcode cart and product pages.

In #3593, we will investigate whether we can (and want to) make express checkout work even when cards are not enabled. This PR is focused only on avoiding presenting a broken payment method to the customer.

## Testing instructions

**Test for wp-admin changes**
1. As a merchant, disable `Credit card / debit card` as a payment method.
2. In `develop`, the Express checkout section still displays Apple Pay/Google Pay as an option. 
3. In this branch, the Express checkout section displays a notice that credit card payments are required for express checkout.

**Test for store-side changes**
1. Checkout and build `develop`.
2.  As a merchant, disable `Credit card / debit card` as a payment method, but enable `Apple Pay/Google Pay`
3. Enable some other payment method, like CashApp.
4. As a shopper, add a product to your cart.
5. Apple Pay/Google Pay will still be available in the block cart and block checkout pages, but using them will lead to `Invalid payment method` error.
6. Checkout and build this branch.
7. Apple Pay/Google Pay should no longer be available. Verify that this is true for both block cart and block checkout.
8. Verify that this is true for both ECE and PRB, by manipulating `WC_Stripe_Feature_Flags::is_stripe_ece_enabled()`.


## Screenshots
Admin-side changes: When `Credit card / debit card` is disabled, `Apple Pay/Google Pay` is hidden, similar to how it Link already behaves.
<img width="500" alt="Screenshot 2024-11-06 at 12 10 39 PM" src="https://github.com/user-attachments/assets/03cc1829-8b5c-4046-b701-4486d34307a8">

---


**Post merge**
